specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2019-10-01-preview/WebTestResults_API.json

```diff
@@ -0,0 +1,976 @@
+{
+	"swagger": "2.0",
+	"info": {
+		"title": "MonitorManagementClient",
+		"x-ms-code-generation-settings": {
+			"name": "MonitorManagementClient"
+		},
+		"version": "2019-10-01-preview"
+	},
+	"host": "management.azure.com",
+	"schemes": [
+		"https"
+	],
+	"consumes": [],
+	"produces": [
+		"application/json"
+	],
+	"security": [
+		{
+			"azure_auth": [
+				"user_impersonation"
+			]
+		}
+	],
+	"securityDefinitions": {
+		"azure_auth": {
+			"type": "oauth2",
+			"authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+			"flow": "implicit",
+			"description": "Azure Active Directory OAuth2 Flow",
+			"scopes": {
+				"user_impersonation": "impersonate your user account"
+			}
+		}
+	},
+	"paths": {
+		"/subscriptions/{subscriptionId}/resoucegroups/{resourceGroupName}/providers/microsoft.insights/webtests/{webtestName}/popbrs/testresultfile": {
+			"get": {
+				"tags": [
+					"FindWebTestResult"
+				],
+				"operationId": "testresultfile_brs",
+				"description": "Returns a file test result for the matching test",
+				"parameters": [
+					{
+						"$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+					},
+					{
+						"$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+					},
+					{
+						"$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+					},
+					{
+						"$ref": "#/parameters/WebtestNameParameter"
+					},
+					{
+						"$ref": "#/parameters/PosixTimeStampParameter"
```

Comment:
Instead of many APIs (one for each format) with a bunch of query parameters does it make sense to have 1 POST API (i.e. webtests/{testName}/generateTestResultsFile) that takes the parameters and the type of test results file to generate?  It would greatly simplify this API